homepage/models.py

```diff
@@ -125,10 +184,54 @@ def get_elective_courses(courses):
         # gets all courses that are electives
         return courses.filter(mandatory=False)
 
-    # --- returns if course has Prerequisites
-    def has_preqs(self):
-        return Prerequisites.does_course_have_prerequisites(self)
+    @staticmethod
+    def get_courses_with_preqs(courses):
+        # gets all courses that have prerequisites
+        have_preqs = []
+        for course in courses:
+            if course.has_preqs():
+                have_preqs.append(course.course_id)
+        return courses.filter(course_id__in=have_preqs)
+
+    @staticmethod
+    def get_courses_without_preqs(courses):
+        # gets all courses that have prerequisites

```

Comment:
need to fix the comment here to gets all courses without prerequisites (or remove both comments if you agree)

Reply:
fixed comment